### PR TITLE
{Service Connector} Fix #24213: `az spring connection create mysql-flexible`: Resolve dependency conflict with `interactive` extension

### LINKED ISSUE #24213: `az interactive` fails to start: ModuleNotFoundError: No module named 'prompt_toolkit.interface'
## Describe the bug

**Command Name**
`az interactive`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
No module named 'prompt_toolkit.interface'
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/homebrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/command_modules/interactive/custom.py", line 41, in start_shell
    azext_interactive.start_shell(cmd, style=style)
  File "/Users/davide.agnello/.azure/cliextensions/interactive/azext_interactive/__init__.py", line 10, in start_shell
    from .azclishell.app import AzInteractiveShell
  File "/Users/davide.agnello/.azure/cliextensions/interactive/azext_interactive/azclishell/app.py", line 33, in <module>
    from prompt_toolkit.interface import Application, CommandLineInterface
ModuleNotFoundError: No module named 'prompt_toolkit.interface'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az interactive`

## Environment Summary
```
macOS-12.5.1-arm64-arm-64bit, Darwin 21.6.0
Python 3.10.8
Installer: HOMEBREW

azure-cli 2.41.0

Extensions:
interactive 0.4.5

Dependencies:
msal 1.20.0b1
azure-mgmt-resource 21.1.0b1
```

### PR DESCRIPTION
**Pull request by xfz11**
**Related command**

**Description**

Fix #24213 by replacing `mycli` with `PyMySQL`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Fix dependency conflict issue

File: src/azure-cli/setup.py [hunk 1]
```diff
@@ -141,9 +141,9 @@
     'fabric~=2.4',
     'javaproperties~=0.5.1',
     'jsondiff~=2.0.0',
-    'mycli~=1.22.2',
     'packaging>=20.9,<22.0',
     'PyGithub~=1.38',
+    'PyMySQL==1.0.2',
```
Comment (bebound) [inline src/azure-cli/setup.py:146]:
```suggestion
    'PyMySQL~=1.0.2',
```
We usually use compatible release in `setup.py`

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_credential_free.py [hunk 2]
```diff
@@ -590,7 +590,11 @@ def get_create_query(self, client_id):
             "select * from pgaadauth_create_principal_with_oid('{0}', '{1}', 'ServicePrincipal', false, false);".format(
                 self.aad_username, client_id),
             'GRANT ALL PRIVILEGES ON DATABASE "{0}" TO "{1}";'.format(
-                self.dbname, self.aad_username)]
+                self.dbname, self.aad_username),
+            'GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "{}";'.format(
+                self.aad_username),
+            'GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO "{}";'.format(
+                self.aad_username)]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_credential_free.py:597]: Is this related to the fix?

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_credential_free.py [hunk 3]
```diff
@@ -215,7 +215,7 @@ def create_aad_user_in_mysql(self, connection_kwargs, query_list):
                             for d in installed_packages)
         if not pym_installed:
             import pip
-            pip.main(['install', 'mycli'])
+            pip.main(['install', 'pymysql'])
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_credential_free.py:218]: `mycli` internally depends on `PyMySQL`:

https://github.com/dbcli/mycli/blob/bc9492a2d425c4d4908c1bfc97c316739a284b9f/setup.py#L27

```py
install_requirements = [
    'click >= 7.0',
    # Temporary to suppress paramiko Blowfish warning which breaks CI.
    # Pinning cryptography should not be needed after paramiko 2.11.0.
    'cryptography == 36.0.2',
    # 'Pygments>=1.6,<=2.11.1',
    'Pygments>=1.6',
    'prompt_toolkit>=3.0.6,<4.0.0',
    'PyMySQL >= 0.9.2',
```

I guess that's why `mycli` can be replaced by `PyMySQL` here.

I am still wondering why `mycli` was chosen in lieu of `PyMySQL` initially. Any functionality from `mycli` is used?

**Review by jiasli — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]
